src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.CoreCLR.cs

```diff
@@ -163,49 +164,56 @@ public override int GetHashCode()
             if (IsGenericMethod)
                 return ValueType.GetHashCodeOfPtr(m_handle);
             else
-                return base.GetHashCode();
+                return m_handle.GetHashCode();
```

Comment:
This should return `m_handle.GetHashCodeOfPtr(m_handle)` in all cases. Also all other places that are hashing the handle should use `GetHashCodeOfPtr` instead of `.GetHashCode()`

Reply:
`GetHashCodeOfPtr` does less work and it should be faster than `base.GetHashCode()`. If it is not faster, we should tweak the implementation to make it master.

Reply:
sure, I noticed `m_handle.GetHashCode()` was also bit faster than `base.GetHashCode()`, but apparently might have been producing it wrong